socless/integrations.py

```diff
@@ -61,6 +65,7 @@ def save_state_results(self, state_name, result, errors={}):
         """
         RESULTS_TABLE = os.environ.get("SOCLESS_RESULTS_TABLE")
         results_table = boto3.resource("dynamodb").Table(RESULTS_TABLE)
+        result = replace_floats_with_decimals(result)
```

Comment:
If we don't do this, what does boto3 default to storing the float?

Reply:
It raises `TypeError Floats are not supported`.

We likely haven't run into this error in the past because we rarely deal with floats in SOCless. So its been a latent bug